tests/main.sh

```diff
@@ -3,6 +3,9 @@
 
 # Always ignore SC2230 ('which' is non-standard. Use builtin 'command -v' instead.)
 export SHELLCHECK_OPTS="-e SC2230 -e SC2039 -e SC2028 -e SC2002 -e SC2005"
+export BOOTSTRAP_REUSE_LOCAL=
```

Comment:
Maybe there is a difference in our shell usage, but in my case running it will always lead to false and thus not reusing the old one.
```
~/golang/src/github.com/juju/juju/tests version_acceptancetest*
❯ juju controllers
Use --refresh option with this command to see the latest information.

Controller       Model                         User   Access     Cloud/Region        Models  Nodes    HA  Version
older*           local-charm-deploy-wrong-git  admin  superuser  localhost/default        4      1  none  2.7-beta1.1  
```

```
~/golang/src/github.com/juju/juju/tests version_acceptancetest*
❯ BOOTSTRAP_REUSE_LOCAL=older ./main.sh cli

==> Checking for dependencies
==> TEST BEGIN: test_cli
==> Checking for dependencies
====> Unable to reuse bootstrapped juju
```

Removing that line works for me